nomad/vault.go

```diff
@@ -408,6 +414,22 @@ func (v *vaultClient) buildClient() error {
 	return nil
 }
 
+// getVaultInitStatus is used to get the init status. It first clears the namespace header, to work around an
+// issue in Vault, then restores it.
+func (v *vaultClient) getVaultInitStatus() (bool, error) {
+	v.l.Lock()
+	defer v.l.Unlock()
+
+	// workaround for Vault behavior where namespace header causes /v1/sys/init (and other) endpoints to fail
+	if ns := v.client.Headers().Get(vaultconsts.NamespaceHeaderName); ns != "" {
+		v.client.SetNamespace("")
+		defer func() {
+			v.client.SetNamespace(ns)
```

Comment:
Perhaps we should have 2 vault Clients: one with the header set and one without? That may make it more obvious in code when we're making requests with or without namespaces.

Locking for the duration of an HTTP call seems particularly dangerous in this code.

Reply:
Just noticed this pattern is only used once, so maybe it's fine?

Reply:
Yeah, I'm not particularly happy about this. It would be nice if the Vault API didn't require this workaround, or if the client would allow the namespace to be overridden on a call-by-call basis. 

A separate http client is cleaner code, thanks.